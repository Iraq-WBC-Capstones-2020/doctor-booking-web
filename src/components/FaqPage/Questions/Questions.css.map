{
    "version": 3,
    "mappings": "AAAA,AAAA,YAAY,CAAC;EACX,aAAa,EAAE,IAAI;EACnB,UAAU,EAAE,GAAG,CAAC,GAAG,CAAC,GAAG,CAAC,mBAAmB;CAC5C;;AACD,AAAA,YAAY,CAAC;EACX,UAAU,EAAE,KAAkB,CAAC,UAAU;EACzC,OAAO,EAAE,wBAAwB;EACjC,OAAO,EAAE,eAAe;EACxB,eAAe,EAAE,wBAAwB;EACzC,WAAW,EAAE,iBAAiB;CAC/B;;AAED,AAAA,YAAY,CAAC,CAAC,CAAC;EACb,MAAM,EAAE,CAAC;EACT,WAAW,EAAE,MAAM;EACnB,UAAU,EAAE,MAAM;EAClB,WAAW,EAAE,MAAM;EACnB,SAAS,EAAE,IAAI;EACf,WAAW,EAAE,IAAI;EACjB,cAAc,EAAE,MAAM;EACtB,KAAK,EAAE,mBAAmB;CAC3B;;AAED,AAAA,aAAa,CAAC;EACZ,KAAK,EAAE,qBAAqB,CAAC,UAAU;CACxC;;AAED,AAAA,YAAY,CAAC,MAAM,CAAC;EAClB,MAAM,EAAE,IAAI;EACZ,OAAO,EAAE,IAAI;CACd;;AAED,AAAA,UAAU,CAAC;EACT,UAAU,EAAE,gBAAgB;EAC5B,OAAO,EAAE,6BAA6B;EACtC,UAAU,EAAE,MAAM;EAClB,WAAW,EAAE,MAAM;EACnB,SAAS,EAAE,IAAI;EACf,WAAW,EAAE,IAAI;EACjB,cAAc,EAAE,GAAG;EAEnB,KAAK,EAAE,mBAAmB;CAC3B",
    "sources": [
        "Questions.scss"
    ],
    "names": [],
    "file": "Questions.css"
}