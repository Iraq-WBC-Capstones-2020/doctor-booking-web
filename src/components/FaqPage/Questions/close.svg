<svg width="36" height="36" viewBox="0 0 36 36" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M24.462 8L18 14.462L11.538 8L9 10.538L15.462 17L9 23.462L11.538 26L18 19.538L24.462 26L27 23.462L20.538 17L27 10.538L24.462 8Z" fill="#F20152" fill-opacity="0.6"/>
<circle cx="18" cy="17" r="15.75" stroke="#F20152" stroke-opacity="0.6" stroke-width="2.5"/>
</svg>

