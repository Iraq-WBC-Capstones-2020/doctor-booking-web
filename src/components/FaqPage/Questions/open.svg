<svg width="36" height="36" viewBox="0 0 36 36" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M24.885 12.4425L18 19.3125L11.115 12.4425L9 14.5575L18 23.5575L27 14.5575L24.885 12.4425Z" fill="black" fill-opacity="0.54"/>
<circle cx="18" cy="17" r="15.75" stroke="black" stroke-opacity="0.54" stroke-width="2.5"/>
</svg>

