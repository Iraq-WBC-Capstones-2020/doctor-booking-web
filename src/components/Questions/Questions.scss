* {
  background: #f9f9fa;
  margin: 0;
}

.qustionCard {
  border-radius: 12px;
  box-shadow: 0px 2px 6px rgba(0, 0, 0, 0.25);
}
.card-header {
  background: rgb(255, 255, 255) !important;
  padding: 1.2rem 2.5rem !important;
  display: flex !important;
  justify-content: space-between !important;
  align-items: center !important;
}

.card-header p {
  margin: 0;
  font-family: Roboto;
  font-style: normal;
  font-weight: normal;
  font-size: 24px;
  line-height: 28px;
  letter-spacing: -0.5px;
  color: rgba(0, 0, 0, 0.62);
}

.openQuestion {
  color: rgba(242, 1, 82, 0.7) !important;
}

.card-header button {
  border: none;
  outline: none;
}

.card-body {
  background: white !important;
  padding: 1rem 3.2rem 1.5rem !important;
  font-style: normal;
  font-weight: normal;
  font-size: 18px;
  line-height: 21px;
  letter-spacing: 1px;

  color: rgba(0, 0, 0, 0.62);
}
