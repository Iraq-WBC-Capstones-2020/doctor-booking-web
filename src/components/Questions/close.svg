<svg width="36" height="36" viewBox="0 0 36 36" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M18 12.44L9 21.44L11.115 23.555L18 16.685L24.885 23.555L27 21.44L18 12.44Z" fill="#F20152" fill-opacity="0.6"/>
<circle cx="17.0525" cy="18" r="14.8553" stroke="#F20152" stroke-opacity="0.6" stroke-width="2.5"/>
</svg>