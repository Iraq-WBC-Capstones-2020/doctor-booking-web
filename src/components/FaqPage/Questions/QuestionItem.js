import React, { useState } from 'react';
import { Card, Accordion } from 'react-bootstrap';
import './Questions.css';
import open from './open.svg';
import close from './close.svg';
import PropTypes from 'prop-types';

function QuestionItem(props) {
  const [isOpen, setIsOpen] = useState(false);

  QuestionItem.propTypes = {
    question: PropTypes.object.isRequired,
    text: PropTypes.string.isRequired,
    answer: PropTypes.string.isRequired,
    id: PropTypes.number.isRequired,
  };

  return (
    <Accordion>
      <div className="qustionCard my-5">
        <Card.Header>
          <p className={isOpen && 'openQuestion'}>{props.question.question}</p>
          <Accordion.Toggle
            eventKey={props.question.id}
            onClick={() => setIsOpen(!isOpen)}
          >
            {' '}
            <img src={isOpen ? close : open} alt="" />
          </Accordion.Toggle>
        </Card.Header>
        <Accordion.Collapse eventKey={props.question.id}>
          <Card.Body>{props.question.answer}</Card.Body>
        </Accordion.Collapse>
      </div>
    </Accordion>
  );
}

export default QuestionItem;
