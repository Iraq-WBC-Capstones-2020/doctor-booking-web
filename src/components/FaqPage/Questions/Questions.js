import React from 'react';
import { Container } from 'react-bootstrap';
import QuestionItem from './QuestionItem';
import { useTranslation } from 'react-i18next';

function Qustions() {
  const { t, i18n } = useTranslation('validation');
  const questions = [
    {
      question: t('validation:FAQ.question1.Q1'),
      answer: t('validation:FAQ.question1.A1'),
      id: 1,
    },
    {
      question: t('validation:FAQ.question2.Q1'),
      answer: t('validation:FAQ.question2.A1'),
      id: 2,
    },
    {
      question: t('validation:FAQ.question3.Q1'),
      answer: t('validation:FAQ.question3.A1'),
      id: 3,
    },
    {
      question: t('validation:FAQ.question4.Q1'),
      answer: t('validation:FAQ.question4.A1'),
      id: 4,
    },
    {
      question: t('validation:FAQ.question5.Q1'),
      answer: t('validation:FAQ.question5.A1'),
      id: 5,
    },
    {
      question: t('validation:FAQ.question6.Q1'),
      answer: t('validation:FAQ.question6.A1'),
      id: 6,
    },

    {
      question: t('validation:FAQ.question7.Q1'),
      answer: t('validation:FAQ.question7.A1'),
      id: 17,
    },
    {
      question: t('validation:FAQ.question8.Q1'),
      answer: t('validation:FAQ.question8.A1'),
      id: 8,
    },

    {
      question: t('validation:FAQ.question9.Q1'),
      answer: t('validation:FAQ.question9.A1'),
      id: 9,
    },
  ];

  return (
    <div>
      <Container className="my-5">
        {questions.map((question) => (
          <QuestionItem key={question.id} question={question} />
        ))}
      </Container>
    </div>
  );
}

export default Qustions;
