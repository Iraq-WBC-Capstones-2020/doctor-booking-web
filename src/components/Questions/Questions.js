import React, { useState } from 'react';
import { Container } from 'react-bootstrap';
import QuestionItem from './QuestionItem';

// const decoratedOnClick = useAccordionToggle(eventKey, onClick);

function Qustions() {
  const [questions] = useState([
    { question: 'how are u', answer: 'a im fine', id: 1 },
    { question: 'how are u', answer: 'a im fine', id: 2 },
    { question: 'how are u', answer: 'a im fine', id: 3 },
  ]);

  return (
    <div>
      <Container className="my-5">
        {questions.map((question) => (
          <QuestionItem key={question.id} question={question} />
        ))}
      </Container>
    </div>
  );
}

export default Qustions;
