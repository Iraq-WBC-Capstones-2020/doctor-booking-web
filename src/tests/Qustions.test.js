import React from 'react';
import Questions from '../components/FaqPage/Questions/Questions';
import renderer from 'react-test-renderer';

test('Questions (array component)  of faq page render correctly', () => {
  const tree = renderer.create(<Questions />).toJSON();
  expect(tree).toMatchSnapshot();
});
