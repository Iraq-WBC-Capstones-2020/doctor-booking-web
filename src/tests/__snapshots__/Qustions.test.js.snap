// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Questions (array component)  of faq page render correctly 1`] = `
<div>
  <div
    className="my-5 container"
  >
    <div
      className="accordion"
    >
      <div
        className="qustionCard my-5"
      >
        <div
          className="card-header"
        >
          <p
            className={false}
          >
            Who can use this website?
          </p>
          <button
            onClick={[Function]}
            type="button"
          >
             
            <img
              alt=""
              src="open.svg"
            />
          </button>
        </div>
        <div
          aria-expanded={null}
          className="collapse"
        >
          <div
            className="card-body"
          >
            The website is for anyone who wants to find a doctor easily
          </div>
        </div>
      </div>
    </div>
    <div
      className="accordion"
    >
      <div
        className="qustionCard my-5"
      >
        <div
          className="card-header"
        >
          <p
            className={false}
          >
            Am I required to create an acount for booking an online appointment?
          </p>
          <button
            onClick={[Function]}
            type="button"
          >
             
            <img
              alt=""
              src="open.svg"
            />
          </button>
        </div>
        <div
          aria-expanded={null}
          className="collapse"
        >
          <div
            className="card-body"
          >
            No need for an account to book an appointment
          </div>
        </div>
      </div>
    </div>
    <div
      className="accordion"
    >
      <div
        className="qustionCard my-5"
      >
        <div
          className="card-header"
        >
          <p
            className={false}
          >
            Can I schedule more than one appointment with a single doctor?
          </p>
          <button
            onClick={[Function]}
            type="button"
          >
             
            <img
              alt=""
              src="open.svg"
            />
          </button>
        </div>
        <div
          aria-expanded={null}
          className="collapse"
        >
          <div
            className="card-body"
          >
            Yes, you can do that
          </div>
        </div>
      </div>
    </div>
    <div
      className="accordion"
    >
      <div
        className="qustionCard my-5"
      >
        <div
          className="card-header"
        >
          <p
            className={false}
          >
            Can I reschedule an online appointment?
          </p>
          <button
            onClick={[Function]}
            type="button"
          >
             
            <img
              alt=""
              src="open.svg"
            />
          </button>
        </div>
        <div
          aria-expanded={null}
          className="collapse"
        >
          <div
            className="card-body"
          >
            Yes, if you have an urgent situation you can reschedule to another time
          </div>
        </div>
      </div>
    </div>
    <div
      className="accordion"
    >
      <div
        className="qustionCard my-5"
      >
        <div
          className="card-header"
        >
          <p
            className={false}
          >
            How will I know if my booking is confirmed?
          </p>
          <button
            onClick={[Function]}
            type="button"
          >
             
            <img
              alt=""
              src="open.svg"
            />
          </button>
        </div>
        <div
          aria-expanded={null}
          className="collapse"
        >
          <div
            className="card-body"
          >
            The secretary will call you to confirm the appoinment
          </div>
        </div>
      </div>
    </div>
    <div
      className="accordion"
    >
      <div
        className="qustionCard my-5"
      >
        <div
          className="card-header"
        >
          <p
            className={false}
          >
            Should I include a name of a doctor when I want to search for a specific specialty?
          </p>
          <button
            onClick={[Function]}
            type="button"
          >
             
            <img
              alt=""
              src="open.svg"
            />
          </button>
        </div>
        <div
          aria-expanded={null}
          className="collapse"
        >
          <div
            className="card-body"
          >
            No, just select the specialty 
          </div>
        </div>
      </div>
    </div>
    <div
      className="accordion"
    >
      <div
        className="qustionCard my-5"
      >
        <div
          className="card-header"
        >
          <p
            className={false}
          >
            What the doctor's consultation fee will be?
          </p>
          <button
            onClick={[Function]}
            type="button"
          >
             
            <img
              alt=""
              src="open.svg"
            />
          </button>
        </div>
        <div
          aria-expanded={null}
          className="collapse"
        >
          <div
            className="card-body"
          >
            It depends on the doctor himself
          </div>
        </div>
      </div>
    </div>
    <div
      className="accordion"
    >
      <div
        className="qustionCard my-5"
      >
        <div
          className="card-header"
        >
          <p
            className={false}
          >
            I am a Doctor and I have done my registration. What should I do next?
          </p>
          <button
            onClick={[Function]}
            type="button"
          >
             
            <img
              alt=""
              src="open.svg"
            />
          </button>
        </div>
        <div
          aria-expanded={null}
          className="collapse"
        >
          <div
            className="card-body"
          >
            We will send you an email to confirm the registration then any user can search and find you
          </div>
        </div>
      </div>
    </div>
    <div
      className="accordion"
    >
      <div
        className="qustionCard my-5"
      >
        <div
          className="card-header"
        >
          <p
            className={false}
          >
            Can I use the app for checking the availability of a doctor 
 without actually booking or scheduling an appointment?
          </p>
          <button
            onClick={[Function]}
            type="button"
          >
             
            <img
              alt=""
              src="open.svg"
            />
          </button>
        </div>
        <div
          aria-expanded={null}
          className="collapse"
        >
          <div
            className="card-body"
          >
            Yes, while you are searching, click on the doctor card that you want to know about him, it will open another page that contains doctor informaition
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
